File: app/src/main/java/com/supercilex/robotscouter/ui/scout/viewholder/CheckboxViewHolder.java
package com.supercilex.robotscouter.ui.scout.viewholder;

import android.view.View;
import android.widget.CheckBox;

public class CheckboxViewHolder extends ScoutViewHolderBase<Boolean, CheckBox> implements View.OnClickListener {
    public CheckboxViewHolder(View itemView) {
        super(itemView);
    }

    @Override
    public void bind() {
        super.bind();
        mName.setChecked(mMetric.getValue());
        mName.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        updateMetricValue(mName.isChecked());
    }
}

File: app/src/main/java/com/supercilex/robotscouter/ui/scout/viewholder/CheckboxViewHolder.kt
package com.supercilex.robotscouter.ui.scout.viewholder

import android.support.annotation.CallSuper
import android.view.View
import android.widget.CheckBox

open class CheckboxViewHolder(itemView: View) :
        ScoutViewHolderBase<Boolean, CheckBox>(itemView),
        View.OnClickListener {
    public override fun bind() {
        super.bind()
        mName.isChecked = mMetric.value
        mName.setOnClickListener(this)
    }

    @CallSuper
    override fun onClick(v: View) = updateMetricValue(mName.isChecked)
}
